compaction/compaction.cc

```diff
@@ -633,6 +652,13 @@ class compaction {
 
     virtual ~compaction() {
     }
+protected:
+    flat_mutation_reader_v2 maybe_filter_sstable_reader(flat_mutation_reader_v2&& reader) const {
+        if (!_owned_ranges) {
```

Comment:
maybe_filtering_*

or perhaps we can just give it a generic name such as make_compaction_sstable_reader, which underlying implementation depends on compaction settings.

Reply:
okay, a generic name makes sense

Reply:
I see you forgot to change the name of this function.

Reply:
sorry, will fix next version

Reply:
renamed to make_compaction_sstable_reader in next version (v3)